builtin/providers/newrelic/data_source_newrelic_application_test.go

```diff
@@ -0,0 +1,48 @@
+package newrelic
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccNewRelicApplication_Basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccNewRelicApplicationConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccNewRelicApplication("data.newrelic_application.app"),
+				),
+			},
+		},
+	})
+}
+
+func testAccNewRelicApplication(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		r := s.RootModule().Resources[n]
+		a := r.Primary.Attributes
+
+		if a["id"] == "" {
+			return fmt.Errorf("Expected to get an application from New Relic")
+		}
+
+		if a["name"] != testAccExpectedApplicationName {
+			return fmt.Errorf("Expected the application name to be: %s, but got: %s", testAccExpectedApplicationName, a["name"])
+		}
+
+		return nil
+	}
+}
+
+// This name must match the constant defined in provider_test.go
+const testAccNewRelicApplicationConfig = `
+data "newrelic_application" "app" {
+	name = "terraform-test-application"
```

Comment:
We should randomise this application name - this would stop any duplicate resource names

An example can be found [here](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_iam_instance_profile_test.go#L160)